### ISSUE TITLE
首先感谢您的分享。您说“这种反向仅计算导数，因此不被视为反向传播。”hinton指出不需要知道具体细节进行求导。这一步操作不是违背了论文中的观点吗？谢谢

### ISSUE THREAD

**Issue by zym1599** [mohammadpz/pytorch_forward_forward#6]
首先感谢您的分享。您说“这种反向仅计算导数，因此不被视为反向传播。”hinton指出不需要知道具体细节进行求导。这一步操作不是违背了论文中的观点吗？谢谢

**Comment by makrout** [comment]
Backpropagation requires gradients to flow back across layers, and hence does not enjoy the locality update feature. Here, gradients, however, don't flow across layers and are used for local updates only. For this reason, the layers have decoupled backwards passes, and this is why [these lines of code](https://github.com/mohammadpz/pytorch_forward_forward/blob/main/main.py#L96-L97) perform this local update by involving the layer cost function. This can be perceived as 1-layer backpropagation and can be substituted with other estimation/optimization methods that do not require gradients. In other words, this code is using local gradients without loss of generality and is not against Hinton's claim.

**Comment by taomanwai** [comment]
Other estimation/optimization methods that do not require gradients <- any suggestions?
